modules/widgets/googleplus-badge.php

```diff
@@ -0,0 +1,309 @@
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'layout' ); ?>">
+				<?php _e( 'Layout', 'jetpack' ); ?>
+				<select name="<?php echo $this->get_field_name( 'layout' ); ?>" id="<?php echo $this->get_field_id( 'layout' ); ?>">
+					<option value="landscape" <?php selected( $instance['layout'], 'landscape' ); ?>><?php _e( 'Landscape', 'jetpack' ); ?></option>
+					<option value="portrait" <?php selected( $instance['layout'], 'portrait' ); ?>><?php _e( 'Portrait', 'jetpack' ); ?></option>
+				</select>
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'theme' ); ?>">
+				<?php _e( 'Theme', 'jetpack' ); ?>
+				<select name="<?php echo $this->get_field_name( 'theme' ); ?>" id="<?php echo $this->get_field_id( 'theme' ); ?>">
+					<option value="light" <?php selected( $instance['theme'], 'light' ); ?>><?php _e( 'Light', 'jetpack' ); ?></option>
+					<option value="dark" <?php selected( $instance['theme'], 'dark' ); ?>><?php _e( 'Dark', 'jetpack' ); ?></option>
+				</select>
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'show_coverphoto' ); ?>">
+				<input type="checkbox" name="<?php echo $this->get_field_name( 'show_coverphoto' ); ?>" id="<?php echo $this->get_field_id( 'show_coverphoto' ); ?>" <?php checked( $instance['show_coverphoto'] ); ?> class="googleplus-badge-only-person googleplus-badge-only-page" />
+				<?php _e( 'Show Cover Photo', 'jetpack' ); ?>
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'show_photo' ); ?>">
+				<input type="checkbox" name="<?php echo $this->get_field_name( 'show_photo' ); ?>" id="<?php echo $this->get_field_id( 'show_photo' ); ?>" <?php checked( $instance['show_photo'] ); ?> class="googleplus-badge-only-community" />
+				<?php _e( 'Show Photo', 'jetpack' ); ?>
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'show_owners' ); ?>">
+				<input type="checkbox" name="<?php echo $this->get_field_name( 'show_owners' ); ?>" id="<?php echo $this->get_field_id( 'show_owners' ); ?>" <?php checked( $instance['show_owners'] ); ?> class="googleplus-badge-only-community" />
+				<?php _e( 'Show Owners', 'jetpack' ); ?>
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'show_tagline' ); ?>">
+				<input type="checkbox" name="<?php echo $this->get_field_name( 'show_tagline' ); ?>" id="<?php echo $this->get_field_id( 'show_tagline' ); ?>" <?php checked( $instance['show_tagline'] ); ?> />
+				<?php _e( 'Show Tag Line', 'jetpack' ); ?>
+			</label>
+		</p>
+
+		<?php
+	}
+
+	function is_valid_googleplus_url( $url ) {
+		return ( FALSE !== strpos( $url, 'plus.google.com' ) ) ? TRUE : FALSE;
+	}
+
+	function filter_int( $value, $default = 0, $max = 0, $min = 0 ) {
```

Comment:
Can you use php's `filter_var()` / `FILTER_VALIDATE_INT` instead of custom function?